src/databricks/labs/dqx/engine.py

```diff
@@ -485,27 +520,47 @@ def apply_checks_and_save_in_table(
             input_config: Input configuration (e.g., table/view or file location and read options).
             output_config: Output configuration (e.g., table name, mode, and write options).
             quarantine_config: Optional configuration for writing invalid records.
+            metrics_config: Optional configuration for writing summary metrics.
             ref_dfs: Optional reference DataFrames used by checks.
         """
         # Read data from the specified table
         df = read_input_data(self.spark, input_config)
 
         if quarantine_config:
             # Split data into good and bad records
-            good_df, bad_df = self.apply_checks_and_split(df, checks, ref_dfs)
+            good_df, bad_df, observation = self.apply_checks_and_split(df, checks, ref_dfs)
             save_dataframe_as_table(good_df, output_config)
             save_dataframe_as_table(bad_df, quarantine_config)
         else:
             # Apply checks and write all data to single table
-            checked_df = self.apply_checks(df, checks, ref_dfs)
+            checked_df, observation = self.apply_checks(df, checks, ref_dfs)
             save_dataframe_as_table(checked_df, output_config)
 
+        if observation and metrics_config:
+            # Create DataFrame with observation metrics - keys as column names, values as data
+            metrics = observation.get
+            metrics_df = self.spark.createDataFrame(
```

Comment:
create private method for this, it is used in `apply_checks_by_metadata_and_save_in_table` as well